File: server/random/BasicRandom.ts
import {AbstractRandom} from "./AbstractRandom";

export class BasicRandom extends AbstractRandom {
    private static instance: BasicRandom;

    private constructor() {
        super();
    }

    public static getInstance(): BasicRandom {
        if (!BasicRandom.instance) {
            BasicRandom.instance = new BasicRandom();
        }
        return BasicRandom.instance;
    }

    public generateInt(min: number, max: number): number {
        return Math.floor(Math.random() * (max - min + 1) + min);
    }
}

File: java-server/src/main/java/com/random/BasicRandom.java
package com.random;

import java.util.Random;

public class BasicRandom extends AbstractRandom {
    private static volatile BasicRandom instance;

    private BasicRandom() {
        if (instance != null) {
            throw new IllegalStateException("Cannot create another instance of SingletonExample.");
        }
    }

    public static BasicRandom getInstance() {
        if (instance == null) {
            synchronized (BasicRandom.class) {
                if (instance == null) {
                    instance = new BasicRandom();
                }
            }
        }
        return instance;
    }

    @Override
    public int generateInt(int min, int max) {
        Random random = new Random();
        return random.nextInt(max - min + 1) + min;
    }

}
